apps/prairielearn/src/ee/middlewares/requireLinkedLtiUser.ts

```diff
@@ -0,0 +1,47 @@
+import { type NextFunction, type Request, type Response } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import { idsEqual } from '../../lib/id.js';
+import { selectLti13InstanceIdentitiesForCourseInstance } from '../models/lti13-user.js';
+
+/**
+ * Middleware to enforce that a user must have accessed the course through LTI
+ * if the LTI instance is configured to require it. This should be run AFTER
+ * the basic authorization middleware that confirms the user has access to the course instance.
+ */
+export default asyncHandler(async (req: Request, res: Response, next: NextFunction) => {
+  // W'll skip the check in several cases:
+  // 1. If the user is impersonating another user. This could be a useful
+  //    escape hatch for instructors.
+  // 2. If the user isn't already enrolled in the course instance. We'll
+  //    allow them proceed through any necessary enrollment steps first.
```

Comment:
Can you help me understand the flow here? What happens if a new user follows a deep link to an assessment, for example? They'll bypass this check, get auto-enrolled, see the page, but then on next click they'll hit this check and need to login via Canvas?

Reply:
Matt and I looked at this together. We observed that this `requireLinkedLtiUser` middleware runs after the `autoEnroll` middleware, which will ensure that any non-instructor, non-impersonated user who is not already enrolled will be enrolled, and the conditions there are equivalent to the other bypass conditions in this file.

In other words: anyone who is hitting this bit of code and isn't already enrolled is someone who would match the other two bypass conditions here.

I removed this check in f8c0490.